eclass/haskell-cabal.eclass

```diff
@@ -528,6 +528,10 @@ cabal_src_compile() {
 				cabal-haddock
 			fi
 		fi
+		# re-run build, to recover stuff that haddock makes disappear for some
+		# reason...thankfully, cabal is smart enough to only re-build the parts
+		# that have been lost, so this isn't too expensive.
+		cabal-build
```

Comment:
The text should be more precise why we need it:
1. reference to a bug
2. symptoms of a failure